python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -148,22 +158,20 @@ def register_launch_commands(self):
                     )
                     continue
 
-            # Download the thumbnail to use as the app's icon.
-            if app_icon:
-                sg_icon = shotgun_data.ShotgunDataRetriever.download_thumbnail(
-                    app_icon, self._tk_app
+            if "create_engine_launcher" in dir(sgtk.platform) and app_engine:
```

Comment:
i am wondering if this should be reorganized so that we would always call the normal `_register_launch_command()` at this stage, and try our new fancy auto detection probing in the case the path is `None`?

Also, let's discuss if `prepare_launch` should just take the path rather than a whole `SoftwareVersion` entity - seems if we make that tweak, things will be a lot simpler in the launch app!